### PR TITLE
Update to googletest 1.12.1

### PR DESCRIPTION
**Pull request by calcmogul**

### CODE REVIEW

**Comment by calcmogul** [comment]
I'm not sure whether the config.gradle entries even matter since we didn't change them during the last upgrade.

**Comment by ThadHouse** [comment]
They don’t. Also if we update to native utils 2023.9.0, all of those versions could actually be removed.

File: config.gradle [hunk 1]
```diff
@@ -4,17 +4,6 @@ nativeUtils.addWpiNativeUtils()
 nativeUtils.withCrossRoboRIO()
 nativeUtils.withCrossLinuxArm32()
 nativeUtils.withCrossLinuxArm64()
-nativeUtils {
-  wpi {
-    configureDependencies {
```
Comment (ThadHouse) [inline config.gradle:9]: You still need these 3 lines. You just don't need the internal version settings lines. Just leave that block empty.

**Review by PeterJohnson — APPROVED** [review]

File: config.gradle (hunk 1 above)
Comment (ThadHouse) [reply config.gradle:9]: I guess its not _technically_ needed, but I want to keep it just in case.

**Review by ThadHouse — APPROVED** [review]

**Pull request merged**